pandas/core/reshape/merge.py

```diff
@@ -274,10 +271,20 @@ def _merger(x, y):
     if left_by is not None and right_by is not None:
         raise ValueError("Can only group either left or right frames")
     elif left_by is not None:
+        if isinstance(left_by, str):
```

Comment:
its possible that we want to integrate these checks much more into: https://github.com/pandas-dev/pandas/blob/master/pandas/core/reshape/merge.py#L1247

but this may require some refactoring. (as a followup if you can)

Reply:
I will take a try.